### DISCUSSION TITLE
Converting Cesium Ion .b3dm files to .glb

### DISCUSSION THREAD

**Discussion by jkronk** [timoore/vsgCs#6] [General]
I know this is a little out of left field, but I was working on calling the Cesium Ion REST API and saving some 3D tilesets to file to disk as a .glb. I am having trouble loading them back into VSG because of the Draco decompression I believe. It looks like VSG doesn't load the .glb into a `vsg::Node` with Draco extensions. 

My plan was to save some .b3dm tiles as .glb to be used offline. When opening vsgCs or similar apps, I planned on having the 3D tiles load in a specific hardcoded location. Do you have any experience saving the .b3dm file from Cesium using the Draco compression, and saving it as a .glb file with those extensions removed? My sample .glb works when loaded into https://gltf-viewer.donmccurdy.com/ which is where I discovered the Draco extensions. It looks like the best option is to use something like tiny_gltf and Draco libraries to modify the binary file to remove those extensions and actually decompress and re-save the file. I wasn't sure if there was something in vsgCs that would allow me to have an easier time with the 3D tiles in an offline mode configuration.

**Comment by timoore** [comment]
Hi,
Assimp does support Draco decompression, but it needs to be built with the option enabled.  VSG won't see any of the original Draco-compressed arrays.

Alternatively, you can take a look at the gltf-loader branch of vsgCs, which uses Cesium-native's facilities to load glTF. In order to deal with Draco, however, you would need a pull request that I made to Cesium Native, https://github.com/CesiumGS/cesium-native/pull/610, which isn't seeing a lot of action.
